src/util/JSONClientSingleton.js

```diff
@@ -17,12 +17,32 @@ class JSONClientSingleton {
     }
     this.throttleTimeout_ = null;
     this.debounceErrors_ = false;
+    this.isConnected = false;
+    this.eventHandler = function(type, opt_data) {};
     singletonInstance = this;
     return singletonInstance;
   }
+  
+  setEventHandler(eventHandler) {
+    this.eventHandler = eventHandler;
+  }
+
+  setServerUrl(url, login, password) {
+    this.serverUrl = url;  
+    this.axiosConfig = {};
+    if (login  && password) {
+        this.axiosConfig.auth = {
+            username: login,
+            password: password
```

Comment:
indent is wrong here (2 extra spaces). 
OK sorry for nitpicking, I'll stop now :)